narwhal/primary/src/proposer.rs

```diff
@@ -289,7 +296,18 @@ impl Proposer {
                 match self.make_header().await {
                     Err(e @ DagError::ShuttingDown) => debug!("{e}"),
                     Err(e) => panic!("Unexpected error: {e}"),
-                    Ok(()) => (),
+                    Ok(digests) => {
+                        let reason = if timer_expired {
+                            "timeout"
+                        } else {
+                            "size_reached"
```

Comment:
just my opinion so feel free to ignore, i think "max_size_reached" could be a little bit clearer?

Reply:
Good point, this is actually not the "max size reached", but rather the "threshold_size_reached". There are two constants here that been used:
* `header_num_of_batches_threshold`: The minimum number of batches that will trigger a new header creation 
* `max_header_num_of_batches`: The maximum number of batch digests included in a header

we are looking to tag basically the first situation